master/buildbot/reporters/gerrit.py

```diff
@@ -334,8 +335,12 @@ def getBuildInfo(build):
                         }
             buildInfoList = sorted([getBuildInfo(build) for build in builds], key=lambda bi: bi['name'])
 
-            result = _handleLegacyResult(self.summaryCB(buildInfoList, Results[buildset['results']],
-                                                        self.master, self.summaryArg))
+            result = yield self.summaryCB(buildInfoList,
```

Comment:
@tardyp - agree! Do you want to do this as part of this PR? Or use this as the example for future PRs?